[2.7] bumped min PHP version to 5.3.9

xelaris opened the pull request:
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.7 |
| Fixed tickets |  |

see symfony/symfony@43c05650

xelaris committed:
bumped min PHP version to 5.3.9

xabbuh commented:
You will also have to update `/contributing/code/patches.rst`, `/cookbook/bundles/best_practices.rst`, `/cookbook/deployment/azure-website.rst` and `/reference/configuration/framework.rst`

xelaris commented:
Thank you for the hint @xabbuh. I have changed the commit, but I'm not sure about the changes I made in `/cookbook/bundles/best_practices.rst` and `reference/configuration/framework.rst` since it's more about the Validation API as the PHP version.

xabbuh commented on cookbook/bundles/best_practices.rst:
```diff
@@ -350,6 +350,13 @@ there are 3 modes, which the user can configure in their project:
+    Starting with Symfony 2.7, the support for the 2.4 API has been
+    dropped and the minimal PHP version required for Symfony was
+    increased to 5.3.9. If your bundles requires Symfony >=2.7, you
+    don't need to take care about the 2.4 API anymore.
```
I think we should only document the `2.5` and the `2.5-BC` modes. The `2.4` mode in fact has no added value (related to #4804).

stof replied:
@xabbuh my plan for Sunday is to drop the distinction between 2.5 and 2.5-BC, moving the BC layer in the 2.5 implementation itself (the only reason why we have separate implementations in 2.5 and 2.6 is because we decided to avoid bumping the Symfony requirement at this time and the BC layer cannot be implemented before 5.3.9. Bumping the requirement allows us to have a much better experience (allowing to have the new API always available while still having the BC layer all the time)

WouterJ commented on contributing/code/patches.rst:
```diff
@@ -14,9 +14,15 @@ Before working on Symfony, setup a friendly environment with the following
+* PHP version 5.3.9 or above;
 * `PHPUnit`_ 4.2 or above.
 
+.. note::
```
wouldn't `.. caution::` be better?

xelaris commented:
I have rebased and removed the change in `reference/requirements.rst` since it was fixed in #4952 in the meantime. I also replaced `.. note::` by `.. caution::` as suggested by @WouterJ. I'm still not sure about the Validation API related changes. Please let me know if I should change something.

WouterJ commented:
I think this one is ready

weaverryan commented:
Thanks Alexander! And we have a bit more work to do in #4804 to make sure the 2.4 API is removed in 2.7 entirely, and we show the configurability of the API as deprecated.

Cheddam commented:
Sorry to necro an old change, but the bump in requirement to PHP 5.3.9 for Symfony 2.7 components continues to cause problems for developers using Guzzle or Debian Squeeze, and I have found no explanation for this change (it wasn't even mentioned in the release notes).

In short: Was there a bug in earlier releases of PHP 5.3 that was causing issues in Symfony?
